client/client-multi/client-c/src/kaa/kaa_notification_manager.c

```diff
@@ -563,10 +588,16 @@ kaa_error_t kaa_add_notification_listener(kaa_notification_manager_t *self, kaa_
 kaa_error_t kaa_remove_notification_listener(kaa_notification_manager_t *self, uint32_t *listener_id)
 {
     KAA_RETURN_IF_NIL2(self, listener_id, KAA_ERR_BADPARAM);
-    return kaa_list_remove_first(self->mandatory_listeners
-                               , &kaa_find_notification_listener_by_id
-                               , listener_id
-                               , &kaa_data_destroy);;
+    KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Going to remove mandatory notification listener id '%u'", *listener_id);
```

Comment:
Please, move to the add and change message to:

KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Removed mandatory notification listener id '%u'", *listener_id);